kasa/smart/smartmodule.py

```diff
@@ -22,17 +22,17 @@
 
 
 def allow_update_after(
-    func: Callable[Concatenate[_T, _P], Awaitable[None]],
-) -> Callable[Concatenate[_T, _P], Coroutine[Any, Any, None]]:
+    func: Callable[Concatenate[_T, _P], Awaitable[dict]],
+) -> Callable[Concatenate[_T, _P], Coroutine[Any, Any, dict]]:
```

Comment:
So I'm not that familiar complex types, but I assume this is correct for cases where we are expecting dict as return? I'm wondering if both of these could/should be either Coroutines or Awaitables?